app/code/Magento/CustomerGraphQl/Model/Resolver/UpdateCustomerEmail.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Resolver;
+
+use Magento\CustomerGraphQl\Model\Customer\ExtractCustomerData;
+use Magento\CustomerGraphQl\Model\Customer\GetCustomer;
+use Magento\CustomerGraphQl\Model\Customer\UpdateCustomerAccount;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\GraphQl\Model\Query\ContextInterface;
+
+/**
+ * Customer email update, used for GraphQL request processing
+ */
+class UpdateCustomerEmail implements ResolverInterface
+{
+    /**
+     * @var GetCustomer
+     */
+    private $getCustomer;
+
+    /**
+     * @var UpdateCustomerAccount
+     */
+    private $updateCustomerAccount;
+
+    /**
+     * @var ExtractCustomerData
+     */
+    private $extractCustomerData;
+
+    /**
+     * @param GetCustomer $getCustomer
+     * @param UpdateCustomerAccount $updateCustomerAccount
+     * @param ExtractCustomerData $extractCustomerData
+     */
+    public function __construct(
+        GetCustomer $getCustomer,
+        UpdateCustomerAccount $updateCustomerAccount,
+        ExtractCustomerData $extractCustomerData
+    ) {
+        $this->getCustomer = $getCustomer;
+        $this->updateCustomerAccount = $updateCustomerAccount;
+        $this->extractCustomerData = $extractCustomerData;
+    }
+
+    /**
+     * Resolves customer email update mutation
```

Comment:
we use "Resolve" imperative format. - you used Resolves as descriptive format
Also add params annotation
also add supress annotation, because not all params are used here:

```
PHP Code Mess has found error(s):
/var/www/html/app/code/Magento/CustomerGraphQl/Model/Resolver/UpdateCustomerEmail.php:58	Avoid unused parameters such as '$field'.
/var/www/html/app/code/Magento/CustomerGraphQl/Model/Resolver/UpdateCustomerEmail.php:60	Avoid unused parameters such as '$info'.
/var/www/html/app/code/Magento/CustomerGraphQl/Model/Resolver/UpdateCustomerEmail.php:61	Avoid unused parameters such as '$value'.
```